.github/workflows/build.yml

```diff
@@ -48,6 +48,28 @@ jobs:
             git diff --name-only origin/$GITHUB_BASE_REF.. | grep -qvE '(\.rst$|^Doc|^Misc)' && echo '::set-output name=run_tests::true' || true
           fi
 
+  check_abi:
+    name: 'Check if the ABI has changed'
+    runs-on: ubuntu-latest
```

Comment:
I'd pin the runner, to ensure that the job runs with at least the same major GCC version every time. Ref. bpo-43382.